Python/Product/Analysis/LanguageServer/Server.Hover.cs

```diff
@@ -76,29 +57,71 @@ public sealed partial class Server {
             }
 
             TraceMessage($"Getting hover for {expr.ToCodeString(tree, CodeFormattingOptions.Traditional)}");
-            var values = analysis.GetValues(expr, @params.position, scope).ToList();
 
-            string originalExpr;
-            if (expr is ConstantExpression || expr is ErrorExpression) {
-                originalExpr = null;
+            // First try values from expression. This works for 
+            // import statements most of the time as well.
+            var index = tree.LocationToIndex(@params.position);
+            var w = new ImportedModuleNameWalker(entry, index, tree);
+            tree.Walk(w);
+
+            IEnumerable<AnalysisValue> values = null;
+            // Perhaps it is an assigned member like 'p' in 'from os import path as p'
+            if (!string.IsNullOrEmpty(w.ImportedType?.Name)) {
+                values = analysis.GetValues(w.ImportedType.Name, @params.position).ToList();
             } else {
-                originalExpr = @params._expr?.Trim();
-                if (string.IsNullOrEmpty(originalExpr)) {
-                    originalExpr = expr.ToCodeString(tree, CodeFormattingOptions.Traditional);
+                // Perhaps it is a member like 'path' in 'from os import path'
+                w.ImportedMembers.FirstOrDefault(m => {
+                    values = analysis.GetValues(m.Name, @params.position).ToList();
+                    return values.Any();
+                });
+            }
+
+            values = values ?? analysis.GetValues(expr, @params.position, scope).ToList();
+            if (values.Any()) {
+                string originalExpr;
+                if (expr is ConstantExpression || expr is ErrorExpression) {
+                    originalExpr = null;
+                } else {
+                    originalExpr = @params._expr?.Trim();
+                    if (string.IsNullOrEmpty(originalExpr)) {
+                        originalExpr = expr.ToCodeString(tree, CodeFormattingOptions.Traditional);
+                    }
+                }
+
+                var names = values.Select(GetFullTypeName).Where(n => !string.IsNullOrEmpty(n)).Distinct().ToArray();
+                var res = new Hover {
+                    contents = GetMarkupContent(
+                        _displayTextBuilder.GetDocumentation(values, originalExpr),
+                        _clientCaps.textDocument?.hover?.contentFormat),
+                    range = exprSpan,
+                    _version = version?.Version,
+                    _typeNames = names
+                };
+                return Task.FromResult(res);
+            }
+
+            if (w.ImportedModules.Any()) {
+                var sb = new StringBuilder();
+                var range = default(Range);
+                foreach (var n in w.ImportedModules) {
+                    if (_analyzer.Modules.TryImport(n.Name, out var modRef)) {
```

Comment:
This one should probably be `TryGetImportedModule` so we aren't modifying the analysis state or triggering background tasks from a hover request.